servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -325,8 +325,17 @@ public EntriesResponse getEntries(String namedRef, EntriesParams params)
                           .name(fromKey(key.getValue()))
                           .type((Type) key.getType())
                           .build())) {
-        entries =
-            filterEntries(s, params.queryExpression()).collect(ImmutableList.toImmutableList());
+        Stream<EntriesResponse.Entry> entriesStream =
+            filterEntries(entryStream, params.queryExpression());
+        if (params.namespaceDepth() != null && params.namespaceDepth() > 0) {
+          entries =
+              entriesStream
+                  .map(e -> truncate(e, params.namespaceDepth()))
+                  .distinct()
+                  .collect(ImmutableList.toImmutableList());
+        } else {
+          entries = entriesStream.collect(ImmutableList.toImmutableList());
```

Comment:
Do we respect `EntriesParams.maxRecords`? Should we document that it does not apply in this case?

Same concern about `pageToken`

Reply:
There is a block of todo above regarding paging. I would rather not modify any of the paging specific stuff in `getEntries` and keep it to the depth token for this PR.